### PR TITLE
SIMSBIOHUB-461: "Missing Field" snackbar fix

### PR DESCRIPTION
**Pull request by KjartanE**
## Links to Jira Tickets

- https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-461

## Description of Changes

- Remove Scroll to error feature
- Remove extra snackbar message when form is submitted successfully
- Remove bug when snackbar shows before submission

## Testing Notes

- Create/Edit on both Project and Survey pages

### CODE REVIEW

File: app/src/components/alert/SnackbarComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { useFormikContext } from 'formik';
+import { useEffect, useState } from 'react';
+
+const SnackbarComponent = () => {
+  const formikProps = useFormikContext<any>();
+  const { errors, submitCount } = formikProps;
+  const [openSnackbar, setOpenSnackbar] = useState({ open: false, msg: '' });
+
+  useEffect(() => {
+    const showSnackBar = (message: string) => {
```
Comment (curtisupshall) [inline app/src/components/alert/SnackbarComponent.tsx]: Could probably do away with this anonymous function and just directly call `setOpenSnackbar`
```javascript
useEffect(() => {
  if (!errors || submitCount <= 0) {
    return;
  }

  setOpenSnackbar({ open: true, message: 'Missing ...' });
}, [errors, submitCount]);
```

File: app/src/components/alert/SnackbarComponent.tsx
Comment (curtisupshall) [inline app/src/components/alert/SnackbarComponent.tsx:1]: Might give this component a more specific name, e.g. `FormikErrorSnackbar`

File: app/src/components/alert/FormikErrorSnackbar.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    setOpenSnackbar({ open: true, msg: 'Missing one or more required fields.' });
+  }, [errors, submitCount, isSubmitting]);
+
+  const closeSnackBar = () =>
+    setOpenSnackbar((currentState) => {
+      return { open: false, msg: currentState.msg };
+    });
+
+  return (
+    <>
```
Comment (NickPhura) [inline app/src/components/alert/FormikErrorSnackbar.tsx]: super nit: dont need the wrapping fragment

File: app/src/components/alert/FormikErrorSnackbar.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  const formikProps = useFormikContext<any>();
+  const { errors, submitCount, isSubmitting } = formikProps;
+  const [openSnackbar, setOpenSnackbar] = useState({ open: false, msg: '' });
+
+  useEffect(() => {
+    if (!Object.keys(errors).length || submitCount <= 0 || isSubmitting) {
+      return;
+    }
+
+    setOpenSnackbar({ open: true, msg: 'Missing one or more required fields.' });
```
Comment (NickPhura) [inline app/src/components/alert/FormikErrorSnackbar.tsx]: Can obv change this later if Jeremy has feelings about it, but does this need to be more generic?

"One or more required fields is invalid."

File: app/src/components/alert/FormikErrorSnackbar.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  const formikProps = useFormikContext<any>();
+  const { errors, submitCount, isSubmitting } = formikProps;
+  const [openSnackbar, setOpenSnackbar] = useState({ open: false, msg: '' });
+
+  useEffect(() => {
+    if (!Object.keys(errors).length || submitCount <= 0 || isSubmitting) {
+      return;
+    }
+
+    setOpenSnackbar({ open: true, msg: 'One or more fields is invalid.' });
```
Comment (curtisupshall) [inline app/src/components/alert/FormikErrorSnackbar.tsx]: super nit: "One or more fields **are** invalid"

**Review by curtisupshall — COMMENTED** [review]
Tested in Openshift, seems to work as expected. ✅

**Review by curtisupshall — APPROVED** [review]
LGTM

**Review by NickPhura — APPROVED** [review]
🧷

**Pull request merged**